presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -131,6 +138,155 @@ public void testAggregation()
                                                         anyTree(tableScan("orders", ImmutableMap.of("totalprice", "totalprice")))))))));
     }
 
+    @Test
+    public void testWindowAfterJoin()
+    {
+        // Window Partition key is a super set of join key.
+        assertDistributedPlan("SELECT rank() OVER (PARTITION BY o.orderstatus, o.orderkey) FROM orders o JOIN lineitem l ON o.orderstatus = l.linestatus",
+                anyTree(
+                        window(windowMatcherBuilder -> windowMatcherBuilder
+                                        .specification(specification(ImmutableList.of("orderstatus", "orderkey"), ImmutableList.of(), ImmutableMap.of()))
+                                        .addFunction(functionCall("rank", Optional.empty(), ImmutableList.of())),
+                                exchange(LOCAL, GATHER,
+                                        project(
+                                                join(INNER, ImmutableList.of(equiJoinClause("orderstatus", "linestatus")),
```

Comment:
Maybe check that join is distributed and check that the input is partitioned on the window function keys. If the join is replicated - this plan is incorrect. 

Also i don't feel strong, but maybe it is worth adding at least a single test case when replicated join is selected?

Reply:
@arhimondr It is fairly easy to check distribution type. I'm not sure about the second part though. Could you clarify what "input" means here and how I could test it? Maybe I'm missing something, but I didn't see an easier way to test it with current framework.